### PR TITLE
Use aws sdk v1 in spark metadata client

### PR DESCRIPTION
**Pull request by arielshaqed**
Really need to use 1.7.4, which is what Hadoop 2.7.7 used.  But that version is **loooooooonnnnggg** gone.  So go to 1.12.140, which a version from 2022-01 (which should alleviate some AWS log4j worries).

In any case depend on the *provided* version, so we get whatever comes with hadoop-aws (which we do need to bring in some cases).

### CODE REVIEW

**Review by talSofer — APPROVED** [review]
How was this tested (a sanity test)?

**Comment by arielshaqed** [comment]
> How was this tested (a sanity test)?

Not yet tested.  Will try to rebase #3180 on top of this, to get a better e2e test.  Also ideally some integration testing :-)

File: clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala [hunk 1]
```diff
@@ -381,18 +388,18 @@ object S3BulkDeleter {
...
-      System.exit(1)
-    }
-    val repo = args(0)
-    val runID = args(1)
-    val region = args(2)
-    val addressesDFLocation = args(3)
-    val spark = SparkSession.builder().getOrCreate()
-    remove(repo, addressesDFLocation, runID, region, spark)
-  }
+  // def main(args: Array[String]): Unit = {
```
Comment (talSofer) [inline clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala:391]: How does this Spark job run without the main function?

Comment (arielshaqed) [reply clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala:391]: There's another "main" function that gets used.  This used to be another execution method, it was never finished -- so I unified the two objects and got rid of this one.

Sorry, will delete the comment.

File: clients/spark/build.sbt [hunk 2]
```diff
@@ -95,13 +96,17 @@ lazy val spark2Type =
   new BuildType("247", scala211Version, "2.4.7", "0.9.8", "2.7.7", "hadoop2-2.0.1")
 lazy val spark3Type =
   new BuildType("301", scala212Version, "3.0.1", "0.10.11", "2.7.7", "hadoop2-2.0.1")
+lazy val spark312Type =
+  new BuildType("312", scala212Version, "3.1.2", "0.10.11", "3.2.1", "hadoop2-2.0.1")
```
[the remark this answers is not in the document]
Comment (arielshaqed) [reply clients/spark/build.sbt:100]: Right. Added a comment, thanks!

**Review by arielshaqed — COMMENTED** [review]
Thanks!

**Pull request merged**